mypy/partially_defined.py

```diff
@@ -328,6 +332,11 @@ def __init__(
         self.loops: list[Loop] = []
         self.try_depth = 0
         self.tracker = DefinedVariableTracker()
+        builtins_mod = names.get("__builtins__", None)
+        if builtins_mod:
+            assert isinstance(builtins_mod.node, MypyFile)
+            for name in builtins_mod.node.names:
+                self.tracker.record_definition(name)
```

Comment:
I wonder if this might degrade performance, since builtins have over 200 definitions, and it looks like we are iterating over all of them for every module. Can you measure the time spent in this loop when performing a self check, for example?

Reply:
I have found a way to deal with this that doesn't involve calling `record_definition` every time! So performance should be unchanged.